src/utilities/data/DataEnums.hpp

```diff
@@ -249,8 +249,72 @@ OPENSTUDIO_ENUM(BuildingSector,
   ((Residential))
 );
 
+
+
+
+/** \class AppGFuelType
+ *  \brief heat for coil for Appendix G system type selections
+ *  \details See the OPENSTUDIO_ENUM documentation in utilities/core/Enum.hpp. The actual
+ *  macro call is:
+ *  \code
+ OPENSTUDIO_ENUM(AppGFuelType,
+  ((Fuel))
+  ((Electric))
+  ((District))
+  ((HeatPump))
+  ((Solar))
+  ((Mixed))
+);
+ *  \endcode */
+OPENSTUDIO_ENUM(AppGFuelType,
+  ((Fuel))
+  ((Electric))
+  ((District))
+  ((HeatPump))
+  ((Solar))
+  ((Mixed))
+  ((Other))
+);
+
+/** \class ComponentType
+ *  \brief What an HVACComponent is meant to do: heating, cooling, both or none
+ *  \details See the OPENSTUDIO_ENUM documentation in utilities/core/Enum.hpp. The actual
+ *  macro call is:
+ *  \code
+ OPENSTUDIO_ENUM(ComponentType,
+  ((Heating))
+  ((Cooling))
+  ((Both))
+  ((None))
+);
+ *  \endcode */
+OPENSTUDIO_ENUM(ComponentType,
+  ((Heating))
+  ((Cooling))
+  ((Both))
+  ((None))
+);
+
 // clang-format on
 
+inline UTILITIES_API AppGFuelType convertFuelTypeToAppG(FuelType fuelType) {
```

Comment:
Seems like this method will never return the AppG Fuel Types 'HeatPump', 'Solar', or 'Mixed' from the data enums above

Reply:
That's actually normal. The idea there is that any IDD field that takes a `\choice` with a list of fuel types could be automatically convert for convenience.

Solar for eg would be directly harcoded where appropriate

https://github.com/NREL/OpenStudio/blob/6b567a46a6898303630c668dcc3ba17775feb30e/src/model/SolarCollectorFlatPlateWater.cpp#L212-L214